azafea/event_processors/endless/metrics/v3/model/__init__.py

```diff
@@ -540,7 +540,10 @@ class ComputerInformation(SingularEvent):
     """
     __tablename__ = 'computer_information_v3'
     __event_uuid__ = '81f303aa-448d-443d-97f9-8d8a9169321c'
-    __payload_type__ = '(uuuua(sqd))'
+    __payload_type__ = (
+        '(uuuua(sqd))',
+        '(uuuua(sqds))',
```

Comment:
So, if you're going to change the client, why not change this to an `a{sv}` like you talked about in the overview? I don't think you'd want to go so far as to just take it literally from the client and put it into the `info` column, but it seems like it would require no further payload type updates in the future.

One advantage there is that if we decide some field being sent by clients is no longer useful, we can just stop processing it here and don't have to worry that older clients keep sending it. On the other hand, you have to do a bit more type checking and/or conversion to ensure the stored JSONB data is in the format expected.

Reply:
> So, if you're going to change the client, why not change this to an `a{sv}` like you talked about in the overview?

Inertia I guess!

> I don't think you'd want to go so far as to just take it literally from the client and put it into the `info` column, but it seems like it would require no further payload type updates in the future.
> 
> One advantage there is that if we decide some field being sent by clients is no longer useful, we can just stop processing it here and don't have to worry that older clients keep sending it. On the other hand, you have to do a bit more type checking and/or conversion to ensure the stored JSONB data is in the format expected.

You're probably right. I'll experiment with how each side looks with that change.

Reply:
Well, now that I looked at the client side changes, I would completely understand doing it this way for expediency. Changing it from a tuple to a dict might be more trouble than it's worth right now.

Reply:
Looking at the submission side again… I think if I were to do something more invasive I would want to make some much bigger changes. For instance, the CPU info is an array, but that array always has exactly 1 element. I would want to send the CPU vendor, family & model. etc. etc.

So I will shelve the idea of further changes and undraft & land both sides of this.